python/pyarrow/tests/test_cuda.py

```diff
@@ -398,11 +401,13 @@ def test_copy_to_host(size):
             dbuf.copy_to_host(buf=buf, position=position, nbytes=nbytes)
 
 
+@pytest.mark.parametrize("dest_ctx", ['same', 'another'])
 @pytest.mark.parametrize("size", [0, 1, 1000])
-def test_copy_from_device(size):
+def test_copy_from_device(dest_ctx, size):
     arr, buf = make_random_buffer(size=size, target='device')
     lst = arr.tolist()
-    dbuf = buf.context.new_buffer(size)
+    dest_ctx = buf.context if dest_ctx == 'same' else global_context1
```

Comment:
Perhaps skip the test if the two contexts are the same regardless? It would help know whether the "two different devices" case actually gets tested (for example I have a single CUDA GPU).

Reply:
Now on a single GPU system the test  labelled as `another` is skipped because it would be exactly the same as the test labelled as `same`.

The only way to know that copying between two different devices works is to create two
buffers on both devices, make a copy, and then check if the destination buffer contains the
expected data. This is all implemented in the test_copy_from_device function.